lib/safe-pg-migrations/plugins/statement_insurer.rb

```diff
@@ -84,6 +84,30 @@ def add_index(table_name, column_name, **options)
       without_timeout { super(table_name, **options) }
     end
 
+    ruby2_keywords def rename_table(table_name, new_name, skip_view: false)
+      # raise 'DDL transaction is disabled' unless in_transaction?
```

Comment:
`in_transaction?` is always `false` in tests. Even if I wrap 
```ruby
calls = record_calls(@connection, :execute) { run_migration }
```
in
```ruby
calls = @connection.transaction do
  record_calls(@connection, :execute) { run_migration }
end
```
`calls` contain `BEGIN;` as the first line, but `in_transaction?` is still `false`.

Reply:
A-ha, so we override [this method](https://github.com/rails/rails/blob/83217025a171593547d1268651b446d3533e2019/activerecord/lib/active_record/migration.rb#L675). Now I get how `warn_useless '`disable_ddl_transaction`'` works 👍 
Makes total sense. Thanks for the clarification.

Reply:
> room for improvement on this topic

Do you mean that disabling the transaction implies that all migrations should be idempotent, including, e.g. data changes. I can think of:
```ruby
MigrationCountry.find_by(name: 'us').update!(name: 'US')
```
that is not idempotent.
With its benefits, the approach is completely fine.

Did you happen to have atomic migrations that open a transaction to be atomic before `rename_table`? I'd love to use the same style, but couldn't find any example.

https://api.rubyonrails.org/classes/ActiveRecord/Migration.html#class-ActiveRecord::Migration-label-Transactional+Migrations tells this is a valid approach:

> Remember that you can still open your own transactions, even if you are in a `Migration` with `self.disable_ddl_transaction!`

It may come as a surprise, mainly because this note is hidden under a spoiler. The only thing I can think of that needs improvement is documentation of this behaviour:
 - strict idempotency
 - making sections of migrations atomic
 - being prepared that migrations can fail en route

I can take a stab at documenting this when done with `rename_table`/`rename_column`.